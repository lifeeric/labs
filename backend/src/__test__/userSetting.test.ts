import * as util from "util";
import axios from "axios";
import { exec } from "child_process";
import path from "path";

const pExec = util.promisify(exec);

const API = "http://localhost:8000/graphql";

describe("UserSettings.ts", () => {
  it("should fetch data", async () => {
    const query = `
query {
    userSetting(id: "5f449b73e2ccbc43aa520488") {
      __typename
      ... on GetUserSettingResult {
        name
        email
        company
      refered{
        company
        name
      }
      }
      ... on Error {
        error
        message }
    }
  }`;

    const response = await axios.post(
      API,
      { query },
      {
        headers: {
          Authorization:
            "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI1ZjQ0OWI3M2UyY2NiYzQzYWE1MjA0ODgiLCJlbWFpbCI6ImFsZXhAdGVzdC5jb20iLCJpYXQiOjE1OTg1MjA1MzAsImV4cCI6MTYwMTExMjUzMH0.iyjvFqB1rS8kfqaeqzNyM1jpr40AOH2booIP_lTS4CU",
        },
      }
    );

    expect(response.data).toMatchSnapshot();
  });
});
