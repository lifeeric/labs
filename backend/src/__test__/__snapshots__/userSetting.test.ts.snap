// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSettings.ts should fetch data 1`] = `
Object {
  "data": Object {
    "userSetting": Object {
      "__typename": "GetUserSettingResult",
      "company": null,
      "email": "alex@test.com",
      "name": "alex",
      "refered": Array [],
    },
  },
}
`;
